### ISSUE TITLE
DataTable: Same event handlers registered multiple times after DataTable.refresh() is called

### ISSUE THREAD

**Issue by movalz** [primefaces/primefaces#3139]
## 1) Environment
- PrimeFaces version: 6.1.2
- Does it work on the newest sources in GitHub? No
- Application server + version: irrelevant
- Affected browsers: irrelevant (all)

## 2) Expected behavior
When JS function DataTable.refresh() is called the different JS event handlers like those bound in Paginator.bindPageLinkEvents() and DataTable.bindCheckboxEvents() are registered twice - the old one stay bound and new listeners which do the same logic are registered. This is harmless in some cases but harmful in others. 

## 3) Actual behavior

Example for harmful cases:
a) in case of a DataTable with a paginator: when clicking one of the pageLinks like "prev" or "next" the same AJAX request is processed multiple times because in bindPaginator() "this.paginator = new PrimeFaces.widget.Paginator(this.cfg.paginator);" registers new event handlers without deregistering the old ones.
b) in case of table with a checkAllToggler the click handler is registered multiple times in bindCheckboxEvents(). If it's an even number (e.g. after calling DataTable.refresh() exactly once) clicking the "togle all" check box seems to doesn't function because it's implicitly clicked twice.
c) in case of sortable columns the click handler on the column headers is registered multiple times in bindSortEvents(). This causes (similar to a)) multiple identical AJAX requests when clicking on a sort header.

There may be more harmful cases. I haven't walked through all bind...() functions.

## 4) Steps to reproduce

for example a)
- open www.primefaces.org/showcase/ui/data/datatable/paginator.xhtml
- open JS console
- type "PrimeFaces.widgets['widget_j_idt117_j_idt118'].refresh(PrimeFaces.widgets['widget_j_idt117_j_idt118'].cfg)"
- jump to "Network" tab of DevTools
- click "next page" link inside the paginator
- in "Network" tab you can see that two AJAX requests were submitted
- type "PrimeFaces.widgets['widget...'].refresh(PrimeFaces.widgets['widget...'].cfg)" again
- in "Network" tab you can see that three AJAX requests were submitted

## 5) Sample XHTML

see www.primefaces.org/showcase/ui/data/datatable/paginator.xhtml

## 6) Sample bean

no special bean necessary

**Comment by tandraschko** [comment]
A PR would be great!

**Comment by movalz** [comment]
I have partial workarounds for the places where I get trouble with this issue but no common solution for it. A PR with my partial workarounds doesn't make much sense, do it?

**Comment by melloware** [comment]
I just ran into this myself.  I think we need an unbindEvents() method on the datatable widget and then in the refresh() method we should call unbindEvents() before the call to super to it call off() on all the events before reregistering new ones.

I was debugging some code and wondering why I sorted a column and it sent 4 ajax requests for 1 sort.  It was because of this issue.  I will see what I can do.
